### PR TITLE
Add typechain for TypeScript contract bindings

### PR DESCRIPTION
**Pull request by alphastorm**

### CODE REVIEW

File: src/hooks/useHistoricalPoolData.ts [hunk 1]
```diff
@@ -123,31 +124,37 @@ export default function useHistoricalPoolData(
...
+        swapContract.filters.RemoveLiquidityOne(null, null, null, null, null),
         deployedBlock,
       )
 
       const removeLiquidityImbalanceHashes = await getEventHashes(
         library,
         account,
         // eslint-disable-next-line new-cap
-        swapContract.filters.RemoveLiquidityImbalance(),
+        swapContract.filters.RemoveLiquidityImbalance(
```
Comment (hammeiam) [inline src/hooks/useHistoricalPoolData.ts:151]: ew

**Review by hammeiam — APPROVED** [review]
lgtm! I'm excited to have this

File: .gitignore [hunk 2]
```diff
@@ -21,3 +21,6 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+
+# typechain
+types/ethers-contracts
```
Comment (penandlim) [inline .gitignore:26]: dumb question - should this folder contain the types generated in `saddle-contracts` repo?

**Review by penandlim — APPROVED** [review]

File: .gitignore (hunk 2 above)
Comment (alphastorm) [reply .gitignore:26]: It's a subset of the generated types as the `saddle-contract` repo, since we only copied some ABIs into `src/constants/abis/` here.

**Pull request merged**